### PR TITLE
kie-issue#958: Quarkus LTS Upgrade to 3.8.4

### PR DESCRIPTION
**Pull request by yesamer**
Related ticket: https://github.com/apache/incubator-kie-tools/pull/2272

Depends on https://github.com/apache/incubator-kie-tools/pull/2277

### CODE REVIEW

**Comment by yesamer** [comment]
` 'dependencies.dependency.version' for org.apache.kie.sonataflow:sonataflow-quarkus-devui:jar is missing.`

I guess we should use ${revision} as version for that dependency

**Comment by tiagobento** [comment]
@yesamer This package doesn't exist anymore. We're making the `devmode` image point to it locally on 
- https://github.com/apache/incubator-kie-tools/pull/2266


... This PR will have to wait for that.

**Review by porcelli — APPROVED** [review]
thank you for bring the last piece of the 3.8.4 upgrade puzzle!

**Comment by yesamer** [comment]
Hi @tiagobento 
Still throws the same error I previously reported.

`'dependencies.dependency.version' for org.apache.kie.sonataflow:sonataflow-quarkus-devui:jar is missing. @ org.kie.kogito:serverless-logic-web-tools-swf-deployment-quarkus-app:${revision}, /home/runner/work/incubator-kie-tools/incubator-kie-tools/packages/serverless-logic-web-tools-swf-deployment-quarkus-app/pom.xml, line 146, column 17`

Is that expected?

File: packages/serverless-logic-web-tools-base-builder-image/env/index.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ module.exports = composeEnv(
         description: "",
       },
       SERVERLESS_LOGIC_WEB_TOOLS__baseBuilderKogitoImageTag: {
-        default: "999-20240417",
+        default: "999-202400502",
```
Comment (rodrigonull) [inline packages/serverless-logic-web-tools-base-builder-image/env/index.js]:
```suggestion
        default: "999-20240502",
```

File: packages/serverless-logic-web-tools-swf-dev-mode-image/env/index.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ module.exports = composeEnv(
         description: "",
       },
       SERVERLESS_LOGIC_WEB_TOOLS__swfDevModeKogitoImageTag: {
-        default: "999-20240417",
+        default: "999-20240502",
```
Comment (rodrigonull) [inline packages/serverless-logic-web-tools-swf-dev-mode-image/env/index.js]: The kogito-swf-devmode image is now part of KIE tools and we no longer build it as part of the weekly jobs. We build it daily as part of the KIE tools daily dev job with the `latest` tag.

**Review by rodrigonull — CHANGES_REQUESTED** [review]

File: packages/serverless-logic-web-tools-base-builder-image/env/index.js (hunk 1 above)
Comment (yesamer) [reply packages/serverless-logic-web-tools-base-builder-image/env/index.js]: Thank you @rodrigonull !!

File: packages/serverless-logic-web-tools-swf-dev-mode-image/env/index.js (hunk 2 above)
Comment (yesamer) [reply packages/serverless-logic-web-tools-swf-dev-mode-image/env/index.js]: @rodrigonull Updated!

**Comment by tiagobento** [comment]
@yesamer There was a missing dependency on `serverless-logic-web-tools-swf-deployment-quarkus-app`. Fixed and pushed.

**Review by rodrigonull — APPROVED** [review]

**Comment by tiagobento** [comment]
Ok! So we should expect seeing this error on this PR's checks:

`
Error:  Failed to execute goal io.quarkus:quarkus-maven-plugin:3.8.4:build (default) on project serverless-logic-web-tools-swf-deployment-quarkus-app: Execution default of goal io.quarkus:quarkus-maven-plugin:3.8.4:build failed: A required class was missing while executing io.quarkus:quarkus-maven-plugin:3.8.4:build: io/quarkus/deployment/builditem/RunTimeConfigurationSourceValueBuildItem
`

@pefernan is aware and could already reproduce this locally. We should have more info tomorrow! Thanks @yesamer and @pefernan !

**Comment by tiagobento** [comment]
@yesamer @pefernan Just remembered we need to update https://github.com/apache/incubator-kie-sandbox-quarkus-accelerator too!

**Comment by pefernan** [comment]
@yesamer let's see if the new artifacts solve the build!

**Review by pefernan — APPROVED** [review]

**Comment by tiagobento** [comment]
Update: this error is caused by a misconfiguration on the data-index-inmemory module. @pefernan will send a PR to `kogito-apps` fixing it and we'll need a new timestamped SNAPSHOT. cc @yesamer @rodrigonull @porcelli

**Comment by tiagobento** [comment]
🤞

**Pull request merged**